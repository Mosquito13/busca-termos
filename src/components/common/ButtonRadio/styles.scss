@import '../../../sass/styles.scss';

.button-radio {
  height: 5rem;
  display: inline-flex;
  border-radius: $radius;
  @include themify() {
    border: 1px solid themed('color-text');
    color: themed('color-text');
  }

  &__option {
    min-width: 7rem;
    padding: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: transparent;
    font-size: inherit;
    font-family: inherit;
    white-space: nowrap;
    line-height: 1rem;
    transition: all 0.2s;
    border: none;
    outline: none;
    cursor: pointer;
    @include themify() {
      color: themed('color-text');
    }

    &:hover {
      @include themify() {
        color: themed('color-base');
        background-color: themed('color-text');
        box-shadow: 0px 0px 5px themed('color-text');
      }
    }

    &--selected {
      @include themify() {
        background-color: themed('color-contrast');
      }
    }
  }

  &__option:not(:last-child) {
    @include themify() {
      border-right: 1px solid themed('color-text');
    }
  }
}
