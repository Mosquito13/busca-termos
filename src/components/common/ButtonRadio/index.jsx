import classNames from 'classnames';
import PropTypes from 'prop-types';

import './styles.scss';

const getOption = (option, value, onChange) => {
  const customProps = {};

  if (option.customContent) {
    customProps.title = option.text;
  }

  const optionClasses = classNames(
    'button-radio__option',
    value === option.value && 'button-radio__option--selected'
  );

  return (
    <button
      key={`option_${option.value}`}
      className={optionClasses}
      {...customProps}
      onClick={() => onChange(option.value)}
    >
      {option.customContent || option.text}
    </button>
  );
};

const ButtonRadio = ({ options, onChange, value }) => {
  return (
    <div className="button-radio">
      {options.map(option => getOption(option, value, onChange))}
    </div>
  );
};

ButtonRadio.propTypes = {
  /**
   * Opcoes do campo
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * Texto exibido dentro da opcao
       * Se for passado um customContent, o text vira tooltip
       */
      text: PropTypes.string,
      /**
       * Conteudo do botao
       * Sobrescreve text
       */
      customContent: PropTypes.node,
      /**
       * Valor quando selecionado
       */
      value: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    })
  ),
  /**
   * Funcao para alterar o valor
   */
  onChange: PropTypes.func,
  /**
   * Valor selecionado
   */
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};

export default ButtonRadio;
