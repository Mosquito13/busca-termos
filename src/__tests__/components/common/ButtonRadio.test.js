import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ButtonRadio from '../../../components/common/ButtonRadio';

const textOptions = [
  {
    value: 1,
    text: 'One'
  },
  {
    value: 2,
    text: 'Two'
  }
];

const customOptions = [
  {
    value: 1,
    text: 'One',
    customContent: <div>{'Custom option one'}</div>
  },
  {
    value: 2,
    text: 'Two',
    customContent: <div>{'Custom option two'}</div>
  },
  {
    value: 3,
    text: 'Three',
    customContent: <div>{'Custom option three'}</div>
  }
];

describe(`${ButtonRadio}`, () => {
  describe('tests', () => {
    let wrapper, onChangeSpy;

    beforeAll(() => {
      onChangeSpy = jest.fn();

      wrapper = shallow(
        <ButtonRadio value={1} onChange={onChangeSpy} options={textOptions} />
      );
    });

    it('should render', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.button-radio')).toHaveLength(1);
      expect(wrapper.find('.button-radio__option')).toHaveLength(2);
    });

    it('should call onChange when click', () => {
      wrapper.find('.button-radio__option').at(1).simulate('click');
      expect(onChangeSpy).toBeCalledTimes(1);
      expect(onChangeSpy).toBeCalledWith(2);
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render with text options', () => {
      const tree = renderer
        .create(
          <ButtonRadio options={textOptions} onChange={jest.fn()} value={1} />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with custom options', () => {
      const tree = renderer
        .create(
          <ButtonRadio options={customOptions} onChange={jest.fn()} value={1} />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
