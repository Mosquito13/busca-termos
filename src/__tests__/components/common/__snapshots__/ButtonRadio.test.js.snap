// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`({
  options,
  onChange,
  value
}) => {
  /* istanbul ignore next */
  cov_vfpwsgv3v().f[2]++;
  cov_vfpwsgv3v().s[8]++;
  return /*#__PURE__*/(
    /* istanbul ignore next */
    (0, _jsxDevRuntime.jsxDEV)(
    /* istanbul ignore next */
    "div",
    /* istanbul ignore next */
    {
      className: "button-radio",
      children: options.map((option) => {
        /* istanbul ignore next */
        cov_vfpwsgv3v().f[3]++;
        cov_vfpwsgv3v().s[9]++;
        return getOption(option, value, onChange);
      })
    },
    /* istanbul ignore next */
    void 0,
    /* istanbul ignore next */
    false,
    /* istanbul ignore next */
    {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 5
    },
    /* istanbul ignore next */
    void 0)
  );
} snapshots should render with custom options 1`] = `
<div
  className="button-radio"
>
  <button
    className="button-radio__option button-radio__option--selected"
    onClick={[Function]}
    title="One"
  >
    <div>
      Custom option one
    </div>
  </button>
  <button
    className="button-radio__option"
    onClick={[Function]}
    title="Two"
  >
    <div>
      Custom option two
    </div>
  </button>
  <button
    className="button-radio__option"
    onClick={[Function]}
    title="Three"
  >
    <div>
      Custom option three
    </div>
  </button>
</div>
`;

exports[`({
  options,
  onChange,
  value
}) => {
  /* istanbul ignore next */
  cov_vfpwsgv3v().f[2]++;
  cov_vfpwsgv3v().s[8]++;
  return /*#__PURE__*/(
    /* istanbul ignore next */
    (0, _jsxDevRuntime.jsxDEV)(
    /* istanbul ignore next */
    "div",
    /* istanbul ignore next */
    {
      className: "button-radio",
      children: options.map((option) => {
        /* istanbul ignore next */
        cov_vfpwsgv3v().f[3]++;
        cov_vfpwsgv3v().s[9]++;
        return getOption(option, value, onChange);
      })
    },
    /* istanbul ignore next */
    void 0,
    /* istanbul ignore next */
    false,
    /* istanbul ignore next */
    {
      fileName: _jsxFileName,
      lineNumber: 32,
      columnNumber: 5
    },
    /* istanbul ignore next */
    void 0)
  );
} snapshots should render with text options 1`] = `
<div
  className="button-radio"
>
  <button
    className="button-radio__option button-radio__option--selected"
    onClick={[Function]}
  >
    One
  </button>
  <button
    className="button-radio__option"
    onClick={[Function]}
  >
    Two
  </button>
</div>
`;
